src/Symfony/Component/Form/Extension/Core/Type/WeekType.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Extension\Core\Type;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\DataTransformer\WeekDateToArrayTransformer;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\Form\ReversedTransformer;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class WeekType extends AbstractType
+{
+    private static $widgets = [
+        'text' => TextType::class,
+        'choice' => ChoiceType::class,
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        if ('single_text' === $options['widget']) {
+            $builder->addViewTransformer(new ReversedTransformer(new WeekDateToArrayTransformer()));
+            $builder->addModelTransformer(new WeekDateToArrayTransformer());
```

Comment:
The model transformer needs to be added depending on the value of the `input` option (and we should be able to do that independent of the `widget` option.